diff --git a/src/scanner.c b/src/scanner.c
--- a/src/scanner.c
+++ b/src/scanner.c
@@ -925,6 +925,6 @@ bool tree_sitter_yaml_external_scanner_scan(void *payload, TSLexer *lexer, cons
 
     if ((valid_symbols[R_SQT_STR_CTN] && is_r && scn_sqt_str_cnt(scanner, lexer, R_SQT_STR_CTN)) ||
-        (valid_symbols[BR_SQT_STR_CTN] && is_br && scn_sqt_str_cnt(scanner, lexer, BR_SQT_STR_CTN))) {
+        (valid_symbols[BR_SQT_STR_CTN] && (is_br || has_nwl) && scn_sqt_str_cnt(scanner, lexer, BR_SQT_STR_CTN))) {
         return true;
     }
 
@@ -1084,7 +1084,7 @@ bool tree_sitter_yaml_external_scanner_scan(void *payload, TSLexer *lexer, cons
             mrk_end(scanner, lexer);
             RET_SYM(R_DQT_STR_END)
         }
-        if (valid_symbols[BR_DQT_STR_END] && is_br) {
+        if (valid_symbols[BR_DQT_STR_END] && (is_br || has_nwl)) {
             adv(scanner, lexer);
             mrk_end(scanner, lexer);
             RET_SYM(BR_DQT_STR_END)
@@ -1119,7 +1119,7 @@ bool tree_sitter_yaml_external_scanner_scan(void *payload, TSLexer *lexer, cons
                 RET_SYM(R_SQT_STR_END)
             }
         }
-        if (valid_symbols[BR_SQT_STR_END] && is_br) {
+        if (valid_symbols[BR_SQT_STR_END] && (is_br || has_nwl)) {
             adv(scanner, lexer);
             if (lexer->lookahead == '\'') {
                 adv(scanner, lexer);
